src/Api/halls/api.js

```diff
@@ -0,0 +1,59 @@
+import axios from '../../config/axios';
+
+export const fetchHalls = async () => {
+  let halls = [];
+  await axios
+    .get('/halls')
+    .then((response) => {
+      if (response.status === 200) {
+        halls = response.data;
+      }
+    })
+    .catch((e) => console.log(e.response.data));
+  return halls;
+};
+
+// eslint-disable-next-line no-return-await
+export const addHall = async (data, setClose) => {
+  await axios
+    .post('/admin/halls', data)
+    .then((response) => {
+      if (response.status === 200) {
+        setClose(true);
+      }
+    })
+    .catch((e) => console.log(e.response.data));
+};
+
+// eslint-disable-next-line no-return-await
+export const editHall = async (id) => await axios
+  .get(`/admin/halls/${id}`)
+  .then((response) => {
+    if (response.status === 200) {
+      return response.data;
+    }
+    return null;
+  })
+  .catch((e) => console.log(e.response.data));
+
+// eslint-disable-next-line no-return-await
+export const updateHall = async (data) => await axios
+  .patch('/admin/halls', data)
+  .then((response) => {
+    if (response.status === 200) {
+      return response;
+    }
+    return false;
+  })
+  .catch((e) => console.log(e.response.data));
+
+// eslint-disable-next-line no-return-await
+export const removeHall = async (id) => await axios
+  .delete(`/admin/halls/${id}`)
+  .then((response) => {
+    if (response.status === 200) {
+      return response.data;
+    }
+    return null;
+  })
+  .catch((e) => console.log(e.response.data));
```

Comment:
`no-return-await` linters is actually a great rule to prevent unfulfilled promises and server crashes when we do API calls, we should avoid returning in async functions